qiskit/optimization/problems/quadratic_program.py

```diff
@@ -1092,6 +1093,65 @@ def from_ising(self,
         self.minimize(constant=offset, linear=linear_terms, quadratic=quadratic_terms)
         offset -= offset
 
+    def is_feasible(self, x: Union[List[float], np.ndarray], detailed: bool = False) -> bool:
+        """Returns whether a solution is feasible or not
+        Args:
+            x: the input result list returned by the optimizer
+            detailed: [not implemented] returns the violations if set to True
```

Comment:
@adekusar-drl What kind of behavior are you thinking of `detailed=True`, returning `Tuple[bool, List]` or anything else?

Reply:
Returning `Union[bool (detailed=False), Tuple[bool, List] (detailed=True)]` might be another option. It looks a bit complicated...

Reply:
Why wouldn't we always return the violations so its always a tuple. User can choose to ignore the detail if they don;t care. I imagine we would put the detail into the optimization result so that if the result is not feasible the detail is there for you to inspect.

Reply:
@woodsp-ibm The problem is that (well, at least as I understood) if we return a tuple every time this means we can not just write `if p.is_feasible(x):` but instead it should be something like `if p.is_feasible(x)[0]` and this looks ugly. Frankly this is the main reason why I don't want to return `details` every time. Correct me if I'm wrong and it is possible to return a tuple and have a nice `if` statement.

Reply:
I agree it does not look so great - you could name the method get_feasibility returning the Tuple and then do a simple is_feasible method that calls the get feasibility and just returns the boolean from the tuple dropping the details. How do you see this used? - I was more thinking it was an 'internal' method that would be called when a result was computed that would check the feasibility and store the details in the result object such that the end user normally just looked there and did not call this method directly themselves  - though of course nothing preventing them

Reply:
If one solves a problem and gets a result with the infeasible status, the next step is to take a look what is violated and then try to understand why it is violated (e.g. no feasible solution, or an incorrect solver is used, etc). So, in general, we may add this detailed information to the base result class, it would fit well. Returning detailed information every time in this aspect does not look that weird and as I wrote in another comment I suggested to pull out the part of the code where the `if p.is_feasible(x)` statement is to a dedicated method in the base optimizer class, so there would be just one such ugly `if` statement in the code. So, after some thought on how to use all of these, your idea looks attractive.

So, a brief summery:
- detailed info returned every time, the method name is to change.
- feasibility check to implement in the base class.
- [optionally] include detailed info to the base result class.

Does this sound reasonable?

Reply:
We can also separate `is_feasible(x, detailed)` into two methods, e.g., `is_feasible(x) -> bool` and `violated_constraints(x) -> List[Constraint]`.